src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridApiTransport.php

```diff
@@ -121,9 +124,33 @@ private function getPayload(Email $email, Envelope $envelope): array
                 continue;
             }
 
+            if ($header instanceof TagHeader) {
+                if (\count($categories) < 10) {
```

Comment:
I would throw an exception if we reach the maximum instead of silently ignoring the tag.